NMCA_001/NMCA_001_script.lua

```diff
@@ -0,0 +1,767 @@
+    'We believe this Power Station is powering the shields in the enemy base. Destroy them.',  -- description
+    'kill',
+    {
+        MarkUnits = true,
+        Requirements = {
+            { Area = 'M3Power', Category = categories.ueb1201, CompareOp = '<=', Value = 0, ArmyIndex = UEF},
+            { Area = 'M3Power', Category = categories.ueb1201, CompareOp = '<=', Value = 0, ArmyIndex = UEF},
+            { Area = 'M3Power', Category = categories.ueb1201, CompareOp = '<=', Value = 0, ArmyIndex = UEF},
+        },
+    })
+
+    table.insert(AssignedObjectives, ScenarioInfo.M3S1)
+
+    ScenarioInfo.M3S2 = Objectives.CategoriesInArea(
+    'secondary',                      -- type
+    'incomplete',                   -- complete
+    'Construct Tech 2 Air Factory',  -- title
+    'Our Tech 2 Air Factories provide the ability to call in orbital strikes. Construct one and use the ability.',  -- description
+    'Build',
+    {
+        MarkUnits = false,
+        MarkArea = false,
+        Requirements =
+        {
+            {Area = 'M3', Category = categories.inb0202, CompareOp = '>=', Value = 1, ArmyIndex = Player1},
+        },
+    })
+
+    table.insert(AssignedObjectives, ScenarioInfo.M3S2)
+
+    ScenarioInfo.M3P1:AddResultCallback(
+    function(result)
+        if(result) then
+            ScenarioFramework.CDRDeathNISCamera(ScenarioInfo.EnemyCommander)
+            M3BaseAI.DisableBase()
+            ScenarioFramework.EndOperationSafety()
+            ScenarioFramework.Dialogue(OpStrings.EnemyDead, PlayerWin, true)
+        end
+    end) 
+
+    ScenarioInfo.M3S1:AddResultCallback(
+    function(result)
+        if(result) then
+            for _, v in ScenarioInfo.Shields do
+                v:ToggleScriptBit('RULEUTC_ShieldToggle')
+            end
+        end
+    end)
+
+    ScenarioInfo.M3S2:AddResultCallback(
+    function(result)
+        if(result) then
+            ScenarioFramework.Dialogue(OpStrings.Tech2AirFactoryBuilt, nil, true)
+            ForkThread(GiveOrbitalBombardmentAbil)
+        end
+    end)  
+end
+
+function M3SetFactoryObjective()
+    ScenarioFramework.Dialogue(OpStrings.M3_ProtectFactory, nil, true)
+
+    ScenarioInfo.M3S1 = Objectives.Protect(
+    'secondary',                      -- type
+    'incomplete',                   -- complete
+    'Protect Captured Factory',  -- title
+    'The captured Factory provides access to enemy units. Protect the Factory.',  -- description
+        {
+            MarkUnits = true,                               -- target
+            Units = {ScenarioInfo.UEFFactory},
+        }
+    )
+    table.insert(AssignedObjectives, ScenarioInfo.M3S1)
+    ScenarioFramework.CreateUnitDeathTrigger(FacDead, ScenarioInfo.UEFFactory)
+end
+
+function TacDead()
+	ScenarioInfo.M2P2:ManualResult(true)
+    ScenarioFramework.Dialogue(OpStrings.M1_Tac_Dead, nil, true)
+	TMLAlive = false
+end
+
+function LaunchTac()
+	if TMLAlive then

```

Comment:
you could use
`if not ScenarioInfo.TacLauncher:IsDead() then` and you would have one less variable